src/test/kotlin/no/nav/amt_enhetsregister/service/EnhetServiceTest.kt

```diff
@@ -0,0 +1,211 @@
+package no.nav.amt_enhetsregister.service
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import no.nav.amt_enhetsregister.client.BronnoysundClient
+import no.nav.amt_enhetsregister.client.Moderenhet
+import no.nav.amt_enhetsregister.client.Underenhet
+import no.nav.amt_enhetsregister.repository.EnhetRepository
+import no.nav.amt_enhetsregister.repository.type.Enhet
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.transaction.TransactionStatus
+import org.springframework.transaction.support.TransactionTemplate
+import java.time.ZonedDateTime
+import java.util.function.Consumer
+
+private const val ORGNR_UNDERENHET = "999000000"
+
+private const val ORGNR_MODERENHET = "777000000"
+
+class EnhetServiceTest {
+	val enhetRepository: EnhetRepository = mockk()
+	val bronnoysundClient: BronnoysundClient = mockk()
+	val kafkaProducerService: KafkaProducerService = mockk()
+	val transactionTemplate: TransactionTemplate = mockk()
+	val enhetService = EnhetService(enhetRepository, bronnoysundClient, kafkaProducerService, transactionTemplate)
+
+	@BeforeEach
+	fun setupMock() {
+		// mock TransactionTemplate
+		every {
+			transactionTemplate.executeWithoutResult(any())
+		} answers { invocation ->
+			val action = invocation.invocation.args[0] as Consumer<TransactionStatus>
+			action.accept(DummyTransactionStatus())
```

Comment:
Jeg tror du kan bruke `SimpleTransactionStatus()` her i steden for å lage en egen dummy.  Hvis ikke jeg overser noe.

Reply:
Bra tips, var på jakt etter akkurat en slik, en.